src/geos.h

```diff
@@ -203,6 +203,6 @@ extern enum ShapelyErrorCode coordseq_from_buffer(GEOSContextHandle_t ctx,
                                                   npy_intp cs2,
                                 GEOSCoordSequence** coord_seq);
 extern int coordseq_to_buffer(GEOSContextHandle_t ctx, const GEOSCoordSequence* coord_seq,
-                              double* buf, unsigned int size, unsigned int dims);
+                              double* buf, unsigned int size, int hasZ, int hasM);
```

Comment:
This can be left for now and further discussed in #1648 as it can change later.

But to clarify my suggestion a bit more, the argument naming convention "foo_bar" is used in `src/*.h` header files, and even skimming header files from [`numpy.get_include()`](https://github.com/numpy/numpy/tree/main/numpy/_core/include/numpy) too. Data types and internal variables are fine to take whatever convention they wish (i.e. "fooBar").